pkg/config/config_test.go

```diff
@@ -31,20 +31,87 @@ var _ = Describe("Config", func() {
 				{
 					Name: clusterIdentity1,
 					Patterns: []string{
-						fmt.Sprintf("^(%s/)?shoot--(?P<project>.+)--(?P<shoot>.+)$", clusterIdentity1),
-						"^namespace:(?P<namespace>[^/]+)$",
+						fmt.Sprintf("^%s/shoot--(?P<project>.+)--(?P<shoot>.+)$", clusterIdentity1),
+						"^shoot--(?P<project>.+)--(?P<shoot>.+)$",
 					},
 				},
 				{
 					Name: clusterIdentity2,
 					Patterns: []string{
-						fmt.Sprintf("^(%s)?shoot--(?P<project>.+)--(?P<shoot>.+)$", clusterIdentity2),
-						"^namespace:(?P<namespace>[^/]+)$",
+						fmt.Sprintf("^(%s/)?shoot--(?P<project>.+)--(?P<shoot>.+)$", clusterIdentity2),
 					},
 				}},
 		}
 	})
 
+	var patternValue = func(prefix string) string {
+		value := fmt.Sprintf("shoot--%s--%s", project, shoot)
+
+		if prefix != "" {
+			value = fmt.Sprintf("%s/%s", prefix, value)
+		}
+		return value
+	}
+
+	DescribeTable("Validation of valid target values",
+		func(currentGardenName string, patternPrefix string, expectedGarden string) {
+			value := patternValue(patternPrefix)
+			expectedPM := &config.PatternMatch{Garden: expectedGarden, Project: project, Shoot: shoot}
+			match, err := cfg.MatchPattern(currentGardenName, value)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(match).To(Equal(expectedPM))
+		},
+		Entry(
+			"should succeed if pattern for value is (potential) non unique pattern and garden is set in current target (unique case / keep garden)",
+			clusterIdentity1,
+			clusterIdentity1,
+			clusterIdentity1),
+		Entry(
+			"should succeed if pattern for value is (potential) non unique pattern and garden is not set in current target (unique case / set garden)",
+			"",
+			clusterIdentity1,
+			clusterIdentity1),
+		Entry(
+			"should succeed if pattern for value is (potential) non unique pattern and other garden is set in current target (unique case / switch garden)",
+			clusterIdentity2,
+			clusterIdentity1,
+			clusterIdentity1),
+		Entry(
+			"should succeed if pattern for value is (potential) non unique pattern and clusterIdentity1 is set in current target (non unique case / keep garden)",
+			clusterIdentity1,
+			"",
+			clusterIdentity1),
+		Entry(
+			"should succeed if pattern for value is (potential) non unique pattern and clusterIdentity2 is set in current target (non unique case / keep garden)",
+			clusterIdentity2,
+			"",
+			clusterIdentity2),
```

Comment:
is there a difference between these two? or could you get rid of one of them?

Reply:
It is important to not change the cluster. I wanted to cover this by testing with both clusters. Depending on the code, one of those two tests could fail but it I cannot for-see which one (as it depends on the code)